src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -3515,12 +3515,15 @@ void PredictSystemLoads(EnergyPlusData &state,
     }
 
     for (int zoneNum = 1; zoneNum <= state.dataGlobal->NumOfZones; ++zoneNum) {
-        state.dataZoneTempPredictorCorrector->zoneHeatBalance(zoneNum).predictSystemLoad(
-            state, ShortenTimeStepSys, UseZoneTimeStepHistory, PriorTimeStep, zoneNum);
-        if (state.dataHeatBal->doSpaceHeatBalance) {
-            for (int spaceNum : state.dataHeatBal->Zone(zoneNum).spaceIndexes) {
+        auto &thisZoneHB = state.dataZoneTempPredictorCorrector->zoneHeatBalance(zoneNum);
+        thisZoneHB.predictSystemLoad(state, ShortenTimeStepSys, UseZoneTimeStepHistory, PriorTimeStep, zoneNum);
+        for (int spaceNum : state.dataHeatBal->Zone(zoneNum).spaceIndexes) {
+            if (state.dataHeatBal->doSpaceHeatBalance) {
                 state.dataZoneTempPredictorCorrector->spaceHeatBalance(spaceNum).predictSystemLoad(
                     state, ShortenTimeStepSys, UseZoneTimeStepHistory, PriorTimeStep, zoneNum, spaceNum);
+            } else if (ShortenTimeStepSys) {
+                state.dataZoneTempPredictorCorrector->spaceHeatBalance(spaceNum).MAT = thisZoneHB.MAT;
+                state.dataZoneTempPredictorCorrector->spaceHeatBalance(spaceNum).airHumRat = thisZoneHB.airHumRat;
```

Comment:
Won't the spaces be able to float to temperatures other than the zone MAT? i.e., shouldn't the space MAT be the last time step space MAT?

Reply:
> Won't the spaces be able to float to temperatures other than the zone MAT? i.e., shouldn't the space MAT be the last time step space MAT?

This is an `else if` after ` if (state.dataHeatBal->doSpaceHeatBalance)` so it only fires if there's no space heat balance active. Yes, this has gotten a bit messy. By moving surfaces, etc. to use space air temp and humrat (again to avoid having ifs all over the place to select space or zone), if space heat balance is not active, the space conditions need to stay in synch with the zone conditions. So any time the zone temps get moved around, the space temps need to be reset. I'd love to come up with a cleaner approach.